### PR TITLE
Integrate HAProxy helper into the cluster

### LINKED ISSUE #20887: Migrate and adapt HAProxy helper to run in the manager
| Epic branch |
|--|
|[epic-20887-migrate-haproxy-helper](https://github.com/wazuh/wazuh/tree/epic-20887-migrate-haproxy-helper)|

## Description

Some months ago, a tool was developed to improve how agents are distributed in a Wazuh cluster. This tool communicates with both the master node API and a HAProxy balancer to perform these two main tasks:
- Balance TCP sessions through agent reconnections so that all managers serve a similar number of clients.
- Add or remove managers from the HAProxy backend automatically.

Now it has been decided that, for maintenance reasons, the tool should be included in the manager. More specifically, in the framework's cluster folder. 

## Issues
- [x] https://github.com/wazuh/wazuh-tools/issues/268
- [x] https://github.com/wazuh/wazuh/issues/20937
- [x] https://github.com/wazuh/wazuh/issues/20938
- [x] https://github.com/wazuh/wazuh/issues/20936
- [x] https://github.com/wazuh/wazuh/issues/21552
- [x] https://github.com/wazuh/wazuh/issues/20939
- [x] https://github.com/wazuh/wazuh/issues/20940
- [x] https://github.com/wazuh/wazuh/issues/22905
- [x] https://github.com/wazuh/wazuh/issues/20941
- [x] https://github.com/wazuh/wazuh-jenkins/issues/6395
- [x] #23471
- [x] #23531

### PR DESCRIPTION
**Pull request by nico-stefani**
|Related issue|
|---|
|#20887|

## Description

This PR closes #20887. Integrates the HAProxy Helper functionality into the Wazuh cluster.

## Logs/Alerts example

Having the helper enabled in the `cluster` section of the `ossec.conf`

```xml
<cluster>
    <name>wazuh</name>
    <node_name>master-node</node_name>
    <node_type>master</node_type>
    <key>9d273b53510fef702b54a92e9cffc82e</key>
    <port>1516</port>
    <bind_addr>0.0.0.0</bind_addr>
    <nodes>
        <node>wazuh-master</node>
    </nodes>
    <hidden>no</hidden>
    <disabled>no</disabled>
    <haproxy_helper>
        <haproxy_disabled>no</haproxy_disabled>
        <haproxy_address>wazuh-proxy</haproxy_address>
        <haproxy_user>haproxy</haproxy_user>
        <haproxy_password>haproxy</haproxy_password>
    </haproxy_helper>
</cluster>
```

```console
root@wazuh-master:/var/ossec# framework/python/bin/python3 framework/scripts/wazuh_clusterd.py -f
2024/05/17 19:18:26 DEBUG: [Cluster] [Main] Removing '/var/ossec/queue/cluster/'.
2024/05/17 19:18:26 DEBUG: [Cluster] [Main] Removed '/var/ossec/queue/cluster/'.
Starting cluster in foreground (pid: 19103)
2024/05/17 19:18:26 INFO: [Local Server] [Main] Serving on /var/ossec/queue/cluster/c-internal.sock
2024/05/17 19:18:26 DEBUG: [Local Server] [Keep alive] Calculating.
2024/05/17 19:18:26 DEBUG: [Local Server] [Keep alive] Calculated.
2024/05/17 19:18:26 INFO: [Master] [Main] Serving on ('0.0.0.0', 1516)
2024/05/17 19:18:26 DEBUG: [Master] [Keep alive] Calculating.
2024/05/17 19:18:26 DEBUG: [Master] [Keep alive] Calculated.
2024/05/17 19:18:26 INFO: [Master] [Local integrity] Starting.
2024/05/17 19:18:26 INFO: [Master] [Local agent-groups] Sleeping 30s before starting the agent-groups task, waiting for the workers connection.
+2024/05/17 19:18:26 INFO: [HAPHelper] [Main] Proxy was initialized
+2024/05/17 19:18:26 INFO: [HAPHelper] [Main] Ensuring only exists one HAProxy process. Sleeping 30s before start...
2024/05/17 19:18:26 INFO: [Master] [Local integrity] Finished in 0.081s. Calculated metadata of 34 files.
2024/05/17 19:18:30 INFO: [Worker] [Main] Connection from ('172.29.0.4', 58782)
2024/05/17 19:18:30 DEBUG: [Worker] [Main] Command received: b'hello'
2024/05/17 19:18:30 INFO: [Worker] [Main] Connection from ('172.29.0.5', 39832)
2024/05/17 19:18:30 DEBUG: [Worker] [Main] Command received: b'hello'
2024/05/17 19:18:34 INFO: [Master] [Local integrity] Starting.
2024/05/17 19:18:35 INFO: [Master] [Local integrity] Finished in 0.004s. Calculated metadata of 34 files.
2024/05/17 19:18:39 DEBUG: [Worker worker2] [Main] Command received: b'syn_i_w_m_p'
2024/05/17 19:18:39 DEBUG: [Worker worker2] [Main] Command received: b'syn_i_w_m'
```

### CODE REVIEW

File: framework/requirements.txt
Comment (Selutario) [inline framework/requirements.txt:45]: This PR will be blocked until the merging of:
- https://github.com/wazuh/wazuh/issues/20748

**Review by Selutario — CHANGES_REQUESTED** [review]
Looks good—only one request.

File: api/test/integration/mapping/integration_test_api_endpoints.json
Comment (nico-stefani) [reply api/test/integration/mapping/integration_test_api_endpoints.json:1]: Updated in 1fc61732b0c690613988911d06ce8770822d6f86

**Review by Selutario — CHANGES_REQUESTED** [review]
Looks good! There are just some failing checks, take a look:
- https://github.com/wazuh/wazuh/pull/23513/checks?check_run_id=25562899876

**Comment by nico-stefani** [comment]
We validate the proper function of the RPM build in https://github.com/wazuh/wazuh/actions/runs/9319858112

**Review by Selutario — APPROVED** [review]
LGTM. There are some failed checks related to RPM build, but we did not find the reason. Core team ran the test [here](https://github.com/wazuh/wazuh/actions/runs/9319858112/job/25655320373) as @nico-stefani mentioned and it was successful so I'm mergintg this. 

Please, keep an eye on this in case we need to open an issue to fix any of the problems with those files @nico-stefani.

**Pull request merged**
